Dynamic modals

zainniazi opened the pull request:
feature request: dynamic modal with auto generated sections (header, body, footer)

I would like to suggest a alteration in the modal service which can allow us to generate header and footer dynamically by passing parameters to BsModalRef or ModalOptions.

I have already created such modification which doesn't effect the original implementation we just have to pass some parameters to switch to the dynamic model using existing ModalContainerComponent.

Now we can just do this to get desired result.


let modalRef = this.modalService.show(XyzComponent, {
      header: "Bye",
      complete: true
    });
    modalRef.setButtons([{ text: "Hello", cssClass: "btn btn-primary" },
    { text: "Bye", cssClass: "btn btn-warning", click: () => modalRef.hide() }])
  }


header is the title for modal, we can even choose to show the close button and we can use modalRef inside component or in the parent presenting the modal to set buttons
Closes #5293

zainniazi committed:
Dynamic modals

zainniazi committed:
show/hide close button option in ModalOptions

ludmilanesvitiy commented:
@zainniazi Thank you for your time to contribute. But need to add a few changes:
- fix lint errors
- fix existed tests
- cover your changes by unit-tests
- add DEMOs

zainniazi committed:
Fixing text based popups

zainniazi committed:
Fixed build

zainniazi committed:
Merge branch 'development' into feature-dynamic-modal

zainniazi committed:
Function fixes

zainniazi committed:
Lint changes

zainniazi committed:
Fixing Interface name

zainniazi committed:
Fixed lint issues

zainniazi committed:
Fixed lint warning of line length

zainniazi committed:
Fixes and Documentation

zainniazi committed:
Aligning constructor

zainniazi committed:
Remove unused // tslint:disable-next-line:no-any

zainniazi commented:
@ludmilanesvitiy Thanks for the response, I have added the things requested
- fixed lint errors
- fixed existing tests
- covered changes by unit-tests
- added DEMOs

I hope you can guide what else is needed, maybe you guys can suggest me to improve its structure.

zainniazi committed:
Merge branch 'development' into feature-dynamic-modal

zainniazi commented:
Hello @Domainv,
Can you please have a look at the pull request and suggest some changes ?

Domainv commented:
@zainniazi 
Sorry, but I not sure why we need to make overweight of component if we already can pass any component inside and do the same inside the component?

https://valor-software.com/ngx-bootstrap/#/modals#service-component

zainniazi commented:
@Domainv 
The idea behind this implementation was to increase the dynamic behavior of Modals and Improve the re-usability of components.

For Instance:
Right now if i have a component which is to be displayed in a modal and as well as in a normal parent; we can't use a single component, we have to create two components or one component with just a wrapper around.

Making header and footer dynamic and having rendered the whole modal automatically increases the reuse-ability and of course cleaner internal components, templates and even string popup messages.

Do let me know what do you think.

Thanks

zainniazi committed:
Merge branch 'development' into feature-dynamic-modal

zainniazi committed:
Merge branch 'development' into feature-dynamic-modal

daniloff200 committed:
Merge branch 'development' into feature-dynamic-modal

valorkin committed:
Merge branch 'development' into feature-dynamic-modal

daniloff200 committed:
Merge branch 'development' into feature-dynamic-modal

daniloff200 committed:
Merge branch 'development' into feature-dynamic-modal
